### PR TITLE
Fixing SQL Injection Warnings in DatabaseJobHistoryStore (Issue #969)

### PR DESCRIPTION
**Pull request by kadaan**

### CODE REVIEW

**Comment by kadaan** [comment]
@sahilTakiar Can you review this please?

File: gobblin-metastore/src/main/java/gobblin/metastore/database/Filter.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Objects;
+
+class Filter {
```
Comment (sahilTakiar) [inline gobblin-metastore/src/main/java/gobblin/metastore/database/Filter.java]: Can you document this class

**Comment by sahilTakiar** [comment]
Since this fixes the FindBugs warnings, can you remove the filters for these classes from `ligradle/findbugs/findbugsExclude.xml`

File: gobblin-metastore/src/main/java/gobblin/metastore/database/Filter.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        if (values != null) {
+            for (String value : values) {
+                statement.setString(index++, value);
+            }
+        }
+        return index;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (sahilTakiar) [inline gobblin-metastore/src/main/java/gobblin/metastore/database/Filter.java]: This is fine too, but I generally recommend using Lombok's `EqualsAndHashCode` annotation

File: gobblin-metastore/src/main/java/gobblin/metastore/database/Filter.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
```
Comment (abti) [inline gobblin-metastore/src/main/java/gobblin/metastore/database/Filter.java]: Please change to use the Apache 2.0 License header without copyright

**Comment by abti** [comment]
@kadaan Can you please resolve the merge issues, update license header and update the PR?

**Comment by kadaan** [comment]
@abti Can you review this?

**Pull request merged**